apps/ouraring/ouraring.star

```diff
@@ -0,0 +1,193 @@
+
+def activityIcon(score, dim):
+    return render.Padding(pad = (0, 0, 0, 0), child = render.Row(children = [render.Image(src = ACTIVITY_ICON), render.Padding(pad = (2, 2, 0, 0), child = render.Text(str(score), font = "tom-thumb", color = "#FFF" if not dim else "#444"))]))
+
+def sleepIcon(score, dim):
+    return render.Padding(pad = (0, 0, 0, 0), child = render.Row(children = [render.Padding(pad = (0, 1, 0, 0), child = render.Image(src = SLEEP_ICON)), render.Padding(pad = (2, 2, 0, 0), child = render.Text(str(score), font = "tom-thumb", color = "#FFF" if not dim else "#444"))]))
+
+def activityView(readiness_scores, activity_scores, sleep_scores):
+    avg = cal_average(activity_scores)
+    points = []
+    for index, score in enumerate(activity_scores):
+        points.append((index, score - avg))
+
+    return render.Column(
+        children = [render.Row(children = [readinessIcon(readiness_scores[-1], True), activityIcon(activity_scores[-1], False), sleepIcon(sleep_scores[-1], True)], main_align = "space_evenly", expanded = True), render.Plot(
+            data = points,
+            width = 64,
+            height = 19,
+            color = "#0f0",
+            color_inverted = "#F33",
+            fill = True,
+        )],
+        main_align = "space_between",
+        expanded = True,
+    )
+
+def readinessView(readiness_scores, activity_scores, sleep_scores):
+    avg = cal_average(readiness_scores)
+    points = []
+    for index, score in enumerate(readiness_scores):
+        points.append((index, score - avg))
+
+    return render.Column(
+        children = [render.Row(children = [readinessIcon(readiness_scores[-1], False), activityIcon(activity_scores[-1], True), sleepIcon(sleep_scores[-1], True)], main_align = "space_evenly", expanded = True), render.Plot(
+            data = points,
+            width = 64,
+            height = 19,
+            color = "#0f0",
+            color_inverted = "#F33",
+            fill = True,
+        )],
+        main_align = "space_between",
+        expanded = True,
+    )
+
+def sleepView(readiness_scores, activity_scores, sleep_scores):
+    avg = cal_average(sleep_scores)
+    points = []
+    for index, score in enumerate(sleep_scores):
+        points.append((index, score - avg))
+
+    return render.Column(
+        children = [render.Row(children = [readinessIcon(readiness_scores[-1], True), activityIcon(activity_scores[-1], True), sleepIcon(sleep_scores[-1], False)], main_align = "space_evenly", expanded = True), render.Plot(
+            data = points,
+            width = 64,
+            height = 19,
+            color = "#0f0",
+            color_inverted = "#F33",
+            fill = True,
+        )],
+        main_align = "space_between",
+        expanded = True,
+    )
+
+def errorView(message):
+    return render.WrappedText(
+        content = message,
+        width = 64,
+        color = "#fff",
+    )
+
+def main(config):
+    if "apikey" not in config:
```

Comment:
Would it be possible to provide a default graphic for the case when no configuration is provided? That would allow the Tidbyt iOS/Android app to show users an illustrative preview for users before they've dug up an API key.